ods_ci/libs/DataSciencePipelinesAPI.py

```diff
@@ -0,0 +1,242 @@
+        self.route = ''
+        while self.route == '':
+            self.route, _ = self.run_oc(f"oc get route -n {project} ds-pipeline-ui-sample --template={{{{.spec.host}}}}")
+            time.sleep(1)
+
+        print(f"DSP route should be working: {self.route}")
+        status = -1
+        count = 0
+        while status != 200 and count < 30:
+            if count > 0:
+                print(f'({count}): HTTP Status: {status}')
+                time.sleep(1)
+            response, status = self.do_get(f'https://{self.route}/apis/v1beta1/runs', headers={
+                'Authorization': f'Bearer {self.sa_token}'
+            })
+            count += 1
+        return status
+
+    @keyword
+    def remove_pipeline_project(self, project):
+        print(f"We are removing the project({project}) because we could run the test multiple times")
+        self.run_oc(f"oc delete project {project} --wait=true --force=true")
+        print("Wait because it could be in Terminating status")
+        while True:
+            project_status, error = self.run_oc(f"oc get project {project} --template={{{{.status.phase}}}}")
+            print(f"Project status: {project_status}")
+            print(f"Error message: {error}")
+            if project_status == '':
+                break
+            time.sleep(1)
+        self.run_oc(f"oc new-project {project}")
+
+    @keyword
+    def create_pipeline_project(self, project):
+        self.run_oc(f"oc new-project {project}")
+
+    @keyword
+    def create_pipeline(self, url_test_pipeline_run_yaml):
+        print("Creating a pipeline from data science pipelines stack")
+        test_pipeline_run_yaml, _ = self.do_get(url_test_pipeline_run_yaml)
+        filename = 'test_pipeline_run_yaml.yaml'
+        with open(filename, 'w', encoding='utf-8') as f:
+            f.write(test_pipeline_run_yaml)
+        with open(filename, 'rb') as f:
+            response, _ = self.do_upload(f'https://{self.route}/apis/v1beta1/pipelines/upload', files={
+                    'uploadfile': f
+                }, headers={
+                    'Authorization': f'Bearer {self.sa_token}'
+                })
+        os.remove(filename)
+        pipeline_json = json.loads(response)
+        pipeline_id = pipeline_json['id']
+        response, status = self.do_get(f'https://{self.route}/apis/v1beta1/pipelines/{pipeline_id}', headers={
+            'Authorization': f'Bearer {self.sa_token}'
+        })
+        assert status == 200
+        assert json.loads(response)['name'] == filename
+        return pipeline_id
+
+    @keyword
+    def create_run(self, pipeline_id):
+        print("Creating the run from uploaded pipeline")
+        response, status = self.do_post(f'https://{self.route}/apis/v1beta1/runs', headers={
+                'Authorization': f'Bearer {self.sa_token}',
+                'Content-Type': 'application/json'
+            }, json={
+                "name": "test-pipeline-run",
```

Comment:
"test-pipeline-run": Should be a parameter, in order to be able to create multiple runs for the same user (also to test behavior of different run names)